firedrake/checkpointing.py

```diff
@@ -288,3 +289,161 @@ def __del__(self):
         if hasattr(self, "comm"):
             free_comm(self.comm)
             del self.comm
+
+
+class HDF5File(object):
+
+    """An object to mimic the DOLFIN HDF5File class.
+
+    This checkpoint object is capable of writing :class:`~.Function`\s
+    to disk in parallel (using HDF5) and reloading them on the same
+    number of processes and a :func:`~.Mesh` constructed identically.
+
+    :arg filename:
+    :arg file_mode: the access mode (one of :data:`~.FILE_READ`,
+         :data:`~.FILE_CREATE`, or :data:`~.FILE_UPDATE`)
+    :arg comm: communicator the writes should be collective
+         over.
+
+    The argument names are carefully chosen to match DOLFIN's HDF5File.
```

Comment:
As above, if we adopt the convention of referring to DOLFIN compatibility all the way through our documentation, we would end up cluttering the large majority of our docstrings with this information, since most Firedrake objects are deliberately compatible with DOLFIN or some other part of FEniCS.